### PR TITLE
Fix non-http link handling

### PR DESCRIPTION
**Pull request by balintbartok**
refs: MBL-14955
affects: Teacher
release note: Fixed crash when opening telephone links from help menu

test plan: see ticket (add a tel link to the help menu)

### CODE REVIEW

**Comment by inst-danger** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true">Warnings</th>
    </tr>
  </thead>
  <tbody><tr>
      <td>:warning:</td>
      <td>

  One or more files are below the minimum test coverage 50%
  </td>
    </tr>
  </tbody>
</table>


Coverage | New % | Master % | Delta
-------- | ----- | -------- | -----
**Canvas iOS** | 92.69% | -- | --
Core/Core/Submissions/GradeSubmission.swift | 0% | -- | --
Core/Core/Extensions/TransactionExtensions.swift | 0% | -- | --

[MBL-14955](https://instructure.atlassian.net/browse/MBL-14955)

#### Affected Apps: Teacher

#### Release Note: 
Fixed crash when opening telephone links from help menu
<p align="right">
  Generated by :no_entry_sign: <a href="http://github.com/danger/danger-js/">dangerJS</a>
</p>

**Comment by inst-danger** [comment]
<details><summary>Student.ipa</summary>
  <a href="https://app.bitrise.io/artifact/55876703/p/78b2406c29e35baa1586c789fb139b3a">
    
  </a>
</details>
<details><summary>Teacher.ipa</summary>
  <a href="https://app.bitrise.io/artifact/55876723/p/581b3afe505a9ba1f0155c58b509558f">
    
  </a>
</details>
<details><summary>Parent.ipa</summary>
  <a href="https://app.bitrise.io/artifact/55876664/p/fdd1723f364ca2141c008d7e8b3c3b39">
    
  </a>
</details>

File: rn/Teacher/ios/Teacher/TeacherAppDelegate.swift [hunk 1]
```diff
@@ -231,7 +231,7 @@ extension TeacherAppDelegate: LoginDelegate, NativeLoginManagerDelegate {
     }
 
     func openExternalURL(_ url: URL) {
-        guard let from = environment.topViewController else {
+        guard let from = environment.topViewController, url.scheme?.hasPrefix("http") == true else {
```
Comment (vanwagonet) [inline rn/Teacher/ios/Teacher/TeacherAppDelegate.swift:234]: good catch

**Review by vanwagonet — APPROVED** [review]
QA+1

**Pull request merged**